api/models/network/updateNetwork.js

```diff
@@ -0,0 +1,62 @@
+import { isEqual, difference, values, some } from 'lodash'
+import setupNetworkAttrs from './setupNetworkAttrs'
+
+export default function updateNetwork (userId, id, params) {
+  if (!userId) throw new Error('updateNetwork called with no userID')
+  if (!id) throw new Error('updateNetwork called with no ID')
+  return setupNetworkAttrs(userId, params).then(attrs =>
+    bookshelf.transaction(transacting =>
+      // NOTE: EnsureLoad not built to work with belongsToMany relations
+      Network.find(id, {withRelated: ['communities', 'moderators']}).then(network => {
+        return network.save(attrs, {patch: true, transacting})
+        .tap(updatedNetwork => afterUpdatingNetwork(updatedNetwork, {params, userId, transacting}))
+      })
+    )
+  )
+}
+
+export function afterUpdatingNetwork (network, opts) {
+  const {
+    params: {
+      community_ids,
+      moderator_ids
+    },
+    transacting
+  } = opts
+  return Promise.all([
+    updateCommunities(network, community_ids && values(community_ids), transacting), // eslint-disable-line camelcase
```

Comment:
Yeah, well actually this was a bug that I hadn't tracked-down and I'm glad you found it again. `community_ids` is actually coming in as an index object (`{1: <someCommunityId>,  2: <anotherCommunityId>, ... }`) at this stage for some reason. I need to go back and track down why.